beacon-chain/rpc/eth/beacon/handlers_pool.go

```diff
@@ -474,12 +480,58 @@ func (s *Server) GetAttesterSlashings(w http.ResponseWriter, r *http.Request) {
 			ss = append(ss, s)
 		} else {
 			httputil.HandleError(w, fmt.Sprintf("unable to convert slashing of type %T", slashing), http.StatusInternalServerError)
-			return
+			return "", nil
 		}
 	}
 	slashings := structs.AttesterSlashingsFromConsensus(ss)
+	return v, slashings
+}
 
-	httputil.WriteJson(w, &structs.GetAttesterSlashingsResponse{Data: slashings})
+// GetAttesterSlashingsV2 retrieves attester slashings known by the node but
+// not necessarily incorporated into any block, supporting both AttesterSlashing and AttesterSlashingElectra.
+func (s *Server) GetAttesterSlashingsV2(w http.ResponseWriter, r *http.Request) {
+	ctx, span := trace.StartSpan(r.Context(), "beacon.GetAttesterSlashingsV2")
+	defer span.End()
+
+	headState, err := s.ChainInfoFetcher.HeadStateReadOnly(ctx)
+	if err != nil {
+		httputil.HandleError(w, "Could not get head state: "+err.Error(), http.StatusInternalServerError)
+		return
+	}
+	v := headState.Version()
+	resp := &structs.GetAttesterSlashingsV2Response{Version: version.String(v)}
+
+	// Retrieve the slashings based on the version
+	sourceSlashings := s.SlashingsPool.PendingAttesterSlashings(ctx, headState, true /* return unlimited slashings */)
+	if v >= version.Electra {
+		// Handle Electra version
+		ss := make([]*eth.AttesterSlashingElectra, 0, len(sourceSlashings))
+		for _, slashing := range sourceSlashings {
+			s, ok := slashing.(*eth.AttesterSlashingElectra)
+			if ok {
+				ss = append(ss, s)
```

Comment:
you are appending where when you are making a list of a set length, shouldn't you be setting it like ss[I] = s

you should probably add a test to make sure